ynr/apps/people/models.py

```diff
@@ -165,6 +166,9 @@ def get_value_html(self):
         if self.value_type == "twitter_username":
             url = format_html("https://twitter.com/{}", self.value)
 
+        if self.value_type == "mastodon_username":
+            url = format_html("https://mastodon.social/@{}", self.value)
```

Comment:
We're going to have to be smarter than this for Mastodon. Accounts can be on any domain name and I expect we can get valid input in at least two formats: `@username@domain.com` or `domain.com/@username`.

Both are valid, but to link to the account we're going to need the latter. This means catching the first format and converting it to a link.

Reply:
I suggest using `urllib.parse.urlparse` for this

Reply:
I'm not sure where the `get_value_html` function is being used in the wider codebase so I don't know how big of an issue this is, but I second that it is not going to return a correct result for anybody using a different Mastodon domain.

Reply:
I'm not sure how this works. I've run the project locally with somebody who isn't on a mastodon.social url and it displays correctly and links properly in their profile, and yet this function is pointed at a specific domain. I'm scratching my head on this one.

Reply:
Okay I've figured it out. When it's processed the Mastodon link, it's then falling through to the next check: `if self.value.startswith("http")` and overwriting the value of `url` with the Masto link saved against the person and undoing what's just been done in `if self.value_type == "mastodon_username":`. Essentially, the Mastodon check is redundant in this iteration (not sure if this is deliberate though!).

If this is on purpose then that's okay, but if not this function and the associated test will need an update.

Reply:
No, we want it to raise a validation error when the domain isn't valid. Can you remove this check from `ynr/apps/people/models.py` L169-170 and try again? 

```
 if self.value_type == "mastodon_username":
            url = format_html("https://mastodon.social/@{}", self.value)
```

Reply:
Thanks for this. I've refactored with fresh eyes, can you try again?